kinds/report.go

```diff
@@ -0,0 +1,98 @@
+//go:build ignore
+// +build ignore
+
+//go:generate go run report.go
+
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"sort"
+
+	"github.com/grafana/codejen"
+	"github.com/grafana/grafana/pkg/kindsys"
+	"github.com/grafana/grafana/pkg/registry/corekind"
+)
+
+func main() {
+	kinds := loadKindMetas()
+
+	sort.Slice(kinds, func(i, j int) bool {
+		return kinds[i].Common().MachineName < kinds[j].Common().MachineName
+	})
+
+	out := elsedie(json.MarshalIndent(kinds, "", "  "))("error generating json output")
+
+	path := filepath.Join(kindsys.DeclParentPath, "report.json")
+	file := codejen.NewFile(path, out, reportJenny{})
+	file.Data = []byte(fmt.Sprintln(string(file.Data))) // new line at the end of the file
```

Comment:
Otherwise seems to be conflicting with pre-commit scripts; I might be wrong here (or got confused with the side-effect of any other tool -i.e. IDE-) but seemed to me that _(somehow)_ a new line is appended and therefore making `codejen.FS.Verify` to fail 🤷🏻 

I guess it's not a big deal, but completely open for suggestions, of course! 🙂

Reply:
huh, weeeeird....

does it work better if you do this?
```suggestion
	file.Data = append(file.Data, byte('\n')))) // new line at the end of the file
```

Reply:
Will try, but probably, yeah!

Reply:
Finally decided to add the file to the `.prettierignore`. Otherwise, it's constantly conflicting with Golang JSON marshaling, not only the `\n` at the end but also the way arrays and array elements are formatted.